src/mpf.rs

```diff
@@ -119,11 +119,11 @@ impl Mpf {
         }
     }
 
-    pub fn get_str(&mut self, n_digits: i32, base: i32, exp: &mut c_long) -> String{
+    pub fn get_str(&self, n_digits: i32, base: i32, exp: &mut c_long) -> String {
         let c_str = CString::new("").unwrap();
         let out;
         unsafe{
-            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &mut self.mpf));
+            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &self.mpf));
```

Comment:
i'm not seeing a quick and easy entry point to calculate the number of digits. the easiest way would be to let gmp_allocate_func do the work.